services/web/server/src/simcore_service_webserver/meta_version_control.py

```diff
@@ -0,0 +1,173 @@
+""" Access to the to version_control add-on
+
+"""
+
+import logging
+from types import SimpleNamespace
+from typing import List, Optional
+
+from aiopg.sa.result import RowProxy
+from models_library.projects import ProjectIDStr
+
+from .projects.project_models import ProjectDict
+from .version_control_changes import compute_workbench_checksum, eval_wcopy_project_id
+from .version_control_db import VersionControlRepository
+from .version_control_errors import UserUndefined
+from .version_control_models import CommitID, TagProxy
+from .version_control_tags import compose_wcopy_project_tag_name
+
+log = logging.getLogger(__name__)
+
+
+class VersionControlForMetaModeling(VersionControlRepository):
+
+    # TODO: eval inheritace vs composition?
+
+    async def get_wcopy_project_id(
+        self, repo_id: int, commit_id: Optional[int] = None
+    ) -> ProjectIDStr:
+        async with self.engine.acquire() as conn:
+            if commit_id is None:
+                commit = await self._get_HEAD_commit(repo_id, conn)
+                assert commit  # nosec
+                commit_id = commit.id
+                assert commit_id
+
+            return await self._fetch_wcopy_project_id(repo_id, commit_id, conn)
+
+    async def get_wcopy_project(self, repo_id: int, commit_id: int) -> ProjectDict:
+        async with self.engine.acquire() as conn:
+            project_id = await self._fetch_wcopy_project_id(repo_id, commit_id, conn)
+            project = await self.ProjectsOrm(conn).set_filter(uuid=project_id).fetch()
+            assert project  # nosec
+            return dict(project.items())
+
+    async def get_project(self, project_id: ProjectIDStr) -> ProjectDict:
+        async with self.engine.acquire() as conn:
+            if self.user_id is None:
+                # TODO: add message
+                raise UserUndefined
+            project = (
+                await self.ProjectsOrm(conn)
+                .set_filter(uuid=str(project_id), prj_owner=self.user_id)
+                .fetch(
+                    [
+                        "type",
+                        "uuid",
+                        "name",
+                        "description",
+                        "thumbnail",
+                        "prj_owner",
+                        "access_rights",
+                        "workbench",
+                        "ui",
+                        "classifiers",
+                        "dev",
+                        "quality",
+                        "published",
+                        "hidden",
+                    ]
+                )
+            )
+            assert project  # nosec
+            project_as_dict = dict(project.items())
+            # FIXME: hack to avoid validation error
+            if "thumbnail" in project_as_dict:
+                project_as_dict["thumbnail"] = project_as_dict["thumbnail"] or ""
```

Comment:
I've already seen some thumbnail conversion in this PR, can't it not be avoided? Looks strange.

Reply:
As stated in the comment above ... this is an ugly hack to avoid a design problem with the project's model. For this very reason I started the module ``models_library.pydantic_models_factory`` that you reviewed last week.

Therefore, unfortunately this has to stay there until that module is fully operative.